pkg/plugins/golang/go_version.go

```diff
@@ -33,10 +33,9 @@ var (
 		major: 1,
 		minor: 13,
 	}
-	go116alpha1 = goVersion{
-		major:      1,
-		minor:      16,
-		prerelease: "alpha1",
+	goVerMax = goVersion{
+		major: 1,
+		minor: 17,
```

Comment:
This would mean that any 1.17 pre-release and any 1.16.* versions would be accepted. We have been intentionally limiting new versions pretty strictly, I don't see any point made to switch this behavior to loosely accepting all the mentioned versions.

Reply:
Ok so adding `alpha.0` to this go version would theoretically make comparison strictly < 1.17 right?

Restricting patch versions seems silly; why do you see this restriction as desirable behavior?

Reply:
I'm fine with the patch versions, just listed the whole implications of the change.

Reply:
```suggestion
	goVerMax = goVersion{
		major:      1,
		minor:      17,
		prerelease: "alpha1",
```

Reply:
Fixed.